packages/react-components/react-table/src/hooks/useSelection.ts

```diff
@@ -0,0 +1,14 @@
+import { GetRowIdInternal, SelectionStateInternal, UseTableOptions } from './types';
+import { useMultipleSelection } from './useMultipleSelection';
+import { useSingleSelection } from './useSingleSelection';
+
+export function useSelection<TItem>(
```

Comment:
IMHO, wouldn't this hook be a bad idea? I mean, ideally it would be better if the user directly called `useMultipleSelection` or `useSingleSelection`.

Reply:
yeah, I don't like it now either, it's a part of this issue to decompose these